nixos/modules/services/networking/firewall.nix

```diff
@@ -337,6 +456,121 @@ in
+    };
+
+    networking.firewall.rules = mkOption {
+      default = [];
+      example = [];
+      type = types.listOf types.optionSet;
+      description = ''
+        Rules for the iptables firewall.
+      '';
+      options = {
+        ipv4Only = mkOption {
+          default = false;
+          type = types.bool;
+          description = ''
+            Force this rule to apply to ipv4 tables only.
+            Otherwise its decided upon the ipaddress and its format if it
+            applies to ipv6 or not.
+          '';
+        };
+        ipv6Only = mkOption {
+          default = false;
+          type = types.bool;
+          description = ''
+            Force this rule to apply to ipv6 tables only.
+            Otherwise its decided upon the ipaddress and its format if it
+            applies to ipv4 or not.
+          '';
+        };
+
+        sourceAddr = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          description = ''
+          '';
+        };
+        destinationAddr = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          description = ''
+          '';
+        };
+
+        sourcePort = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          example = "10000:15000";
+          description = ''
+            A port or portrange (as string) for the --sport option of iptables.
+            TODO: Validate this on protocol in ["tcp" "udp"]
+          '';
+        };
+        destinationPort = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          example = "80:443";
+          description = ''
+            A port or portrange (as string) for the --dport option of iptables.
+            TODO: Validate this on protocol in ["tcp" "udp"]
+          '';
+        };
+
+        fromInterface = mkOption {
+          default = null;
+          type = types.nullOr types.str;
+          description = ''
+            Source interface to match on.
+            Value is the name of the interface, optionally prefixed with ! to
+            match everything _except_ that interface.
+            Wildcards are allowed: eth+ matches all eth-something interfaces.
+            Special value nil equals all interfaces.
+          '';
+        };
+        toInterface = mkOption {
+          default = null;
+          type = types.nullOr types.str;
+          description = ''
+            Source interface to match on.
+            Value is the name of the interface, optionally prefixed with ! to
+            match everything _except_ that interface.
+            Wildcards are allowed: eth+ matches all eth-something interfaces.
+            Special value nil equals all interfaces.
+          '';
+        };
+        target = mkOption {
+          default = null;
+          type = types.nullOr (
+            types.addCheck types.str (
+              v: v == "ACCEPT" || v == "DROP" || v == "REJECT"
```

Comment:
@copumpkin 
> Also, isn't LOG an option sometimes?

What would you think is better:
```nix
rule = [
  { fromInterface = "eth0"; target = "LOG"; log-prefix="bla: "; log-level = "info"; }
  { fromInterface = "eth0"; target = "ACCEPT"; }
];
```
with two rule-entries to explicitly LOG and ACCEPT the connection/packet? Or
```nix
rule = [
  { fromInterface = "eth0"; target = "ACCEPT"; log-prefix="bla: "; log-level = "info"; }
];
```
to have one rule-entry and create two iptables rules with log and accept implicit. Is there a need for logging without any special ACCEPT/DROP/REJECT handling?